framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -1358,6 +1359,29 @@ private void message(
             messageText = messageText.replaceAll("\n", LINE_SEPARATOR);
         }
 
+        // TODO: when #979 is fixed, remove this hack
+        if (this.processingEnv.getOptions() != null
+                && !this.processingEnv
+                        .getOptions()
+                        .containsKey("conservativeUninferredTypeArguments")) {
+
+            if (messageText.contains("INFERENCE FAILED")) {
+                return;
+            } else if (this.processingEnv.getOptions().containsKey("nomsgtext")) {
+                for (Object arg : args) {
+                    if (arg != null && arg instanceof String) {
```

Comment:
The two null checks for `arg` are redundant, as `instanceof` is only true for non-null values.